op-monitorism/transaction_monitor/monitor.go

```diff
@@ -0,0 +1,325 @@
+				Name:      "unauthorized_transactions_total",
+				Help:      "Number of transactions from unauthorized addresses",
+			},
+			[]string{"watch_address", "from"},
+		),
+
+		thresholdExceededTx: m.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: MetricsNamespace,
+				Name:      "threshold_exceeded_transactions_total",
+				Help:      "Number of transactions exceeding allowed threshold",
+			},
+			[]string{"watch_address", "from", "threshold"},
+		),
+
+		ethSpent: m.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: MetricsNamespace,
+				Name:      "eth_spent_total",
+				Help:      "Cumulative ETH spent by address",
+			},
+			[]string{"address"},
+		),
+
+		unexpectedRpcErrors: m.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: MetricsNamespace,
+				Name:      "unexpected_rpc_errors_total",
+				Help:      "Number of unexpected RPC errors",
+			},
+			[]string{"section", "name"},
+		),
+	}
+
+	// Initialize filters and watchers
+	for _, config := range cfg.WatchConfigs {
+		mon.watchConfigs[config.Address] = config
+
+		for _, filter := range config.Filters {
+			switch filter.Type {
+			case ExactMatchCheck:
+				if match, ok := filter.Params["match"].(string); ok {
+					mon.allowedAddrs[common.HexToAddress(match)] = true
+				}
+			case DisputeGameCheck:
+				if factory, ok := filter.Params["disputeGameFactory"].(string); ok {
+					factoryAddr := common.HexToAddress(factory)
+					watcher := &DisputeGameWatcher{
+						factory:      factoryAddr,
+						allowedGames: make(map[common.Address]bool),
+					}
+					mon.factoryWatchers[factoryAddr] = watcher
+				}
+			}
+		}
+	}
+
+	return mon, nil
+}
+
+func (m *Monitor) Run(ctx context.Context) {
+	currentBlock := m.startBlock
+	var err error
+
+	for {
+		select {
+		case <-ctx.Done():
+			m.log.Info("monitor stopping")
+			return
+		default:
+			// Get current block if not specified
+			if currentBlock == 0 {
+				currentBlock, err = m.client.BlockNumber(ctx)
+				if err != nil {
+					m.log.Error("failed to get block number", "err", err)
+					continue
+				}
+			}
+
+			// Update dispute game watchers
+			for _, watcher := range m.factoryWatchers {
+				if err := m.updateDisputeGames(ctx, watcher, currentBlock); err != nil {
+					m.log.Error("failed to update dispute games", "err", err)
+					continue
+				}
+			}
+
+			block, err := m.client.BlockByNumber(ctx, big.NewInt(int64(currentBlock)))
+			if err != nil {
+				m.log.Error("failed to get block", "number", currentBlock, "err", err)
+				m.unexpectedRpcErrors.WithLabelValues("monitor", "getBlock").Inc()
+				continue
+			}
+
+			for _, tx := range block.Transactions() {
+				if tx.To() == nil {
+					continue
+				}
+
+				if config, exists := m.watchConfigs[*tx.To()]; exists {
+					m.processTx(tx, config)
+				}
+			}
+
+			currentBlock++
+		}
+	}
+}
+
+func (m *Monitor) updateDisputeGames(ctx context.Context, watcher *DisputeGameWatcher, currentBlock uint64) error {
```

Comment:
Noted over DM but this can be replaced with a static check so that we don't need to keep a running list and operate watchers.